cpp/src/parquet/file_writer.cc

```diff
@@ -221,10 +241,13 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
   void InitColumns() {
     for (int i = 0; i < num_columns(); i++) {
       auto col_meta = metadata_->NextColumnChunk();
-      const ColumnDescriptor* column_descr = col_meta->descr();
+      Compression::type compressionCodec;
+      int32_t compression_level;
+      DetermineCompressionCodecAndLevel(properties_, col_meta->descr()->path(),
+                                        compressionCodec, compression_level);
       std::unique_ptr<PageWriter> pager =
-          PageWriter::Open(sink_, properties_->compression(column_descr->path()),
-                           col_meta, properties_->memory_pool(), buffered_row_group_);
+          PageWriter::Open(sink_, compressionCodec, compression_level, col_meta,
+                           properties_->memory_pool(), buffered_row_group_);
```

Comment:
Same comments as above re: this code duplication

Reply:
Answered  above.